packages/loader/container-loader/src/connectionManager.ts

```diff
@@ -568,21 +598,37 @@ export class ConnectionManager implements IConnectionManager {
         return true;
     }
 
+    /**
+     * Cancel in-progress connection attempt.
+     */
+    private cancelConnection() {
+        assert(this.pendingConnection !== undefined, "this.pendingConnection is undefined when trying to cancel");
+        this.pendingConnection.abortController.abort();
+        this.pendingConnection = undefined;
+        this.logger.sendTelemetryEvent({ eventName: "ConnectionAttemptCancelled" });
+    }
+
     /**
      * Once we've successfully gotten a connection, we need to set up state, attach event listeners, and process
      * initial messages.
      * @param connection - The newly established connection
      */
-     private setupNewSuccessfulConnection(connection: IDocumentDeltaConnection, requestedMode: ConnectionMode) {
+     private setupNewSuccessfulConnection(
+         connection: IDocumentDeltaConnection,
+         requestedMode: ConnectionMode,
+         abortSignal: AbortSignal,
```

Comment:
Still feels weird to decouple this AbortSignal from `this.pendingConnection`.  Can you remind me why it's necessary? I can't see it glancing at the callers here.

Reply:
Got it, thanks!  See my suggestion in the other thread about removing `AbortController` from `IPendingDetails` and just having an `abort()` function instead.